docs/DeveloperGuide.md

```diff
@@ -345,6 +375,33 @@ to the rest of the features that uses this method. This allows the UI to make us
 3 months fees as well. Although this results in increased coupling, with proper testing in place, we could mitigate the risk 
 as we ensure that changes in the `FeeUtil` method do not unintentionally changes the behaviour of the other feature.
 
+### General
+#### List Feature
+The list feature displays a list of existing students, and a list of sessions of those students in the TutorBuddy application.
+
+##### Implementation
+This feature is facilitated by `ListCommand` which extends `Command`.
+The method `ListCommand#execute` updates the filtered student list, and the filtered session list by calling the method
+`Model#updateFilteredStudentList` exposed in the `Model` interface.
+
+Given below is an example of how the list student
+1. The user executes the list command with the input `list`.
+2. `LogicManager` executes the input and parses the command using `AddressBookParser`.
+3. `AddressBookParser` identifies the correct command and creates a new `ListCommand`.
+4. `AddressBookParser` returns the new `ListCommand` to `LogicManager`.
+5. `LogicManager` executes the `ListCommand`.
+6. `ListCommand` now calls `Model` to update the `filteredStudents` and  to show all students.
+
+The following sequence diagram shows the interactions when user executes the `list` command:
+![ListSequenceDiagram](images/choonwei/ListSequenceDiagram.png)
+
+<div markdown="span" class="alert alert-info">:information_source: **Note:** The lifeline for `DeleteRecurringSessionCommandParser` should end at the destroy marker (X) but due to a limitation of PlantUML, the lifeline reaches the end of diagram.
```

Comment:
Hmmm you might have missed this out by including "The lifeline for `DeleteRecurringSessionCommandParser`" in the list command implementation.

Reply:
Oh god, thanks for catching, will go through my codes again.